source/assets/js/classes/fef-storage.js

```diff
@@ -0,0 +1,116 @@
+export class FefStorage {
+
+    /**
+     * Returns a value if local storage is available and item is set.
+     * If optionalDefaultValue is given, this is returned if local storage is not available or no item is set
+     *
+     * @param key
+     * @param optionalDefaultValue
+     * @returns {*}
+     */
+    static getItem(key, optionalDefaultValue) {
+        if (!this.isLocalStorageAvailable() && optionalDefaultValue) {
+            return optionalDefaultValue;
+        }
+
+        const item = localStorage.getItem(key);
+
+        if (!item && optionalDefaultValue) {
+            return optionalDefaultValue;
+        }
+
+        return item;
+    }
+
+    /**
+     * Returns (safely) a JS object
+     *
+     * @param key
+     * @returns {*}
+     */
+    static getItemJsonParsed(key) {
+        if (!this.hasItem(key)) {
+            return {};
+        }
+
+        try {
+            return JSON.parse(this.getItem(key, '{}'));
+        } catch (e) {
+            return {};
+        }
+    }
+
+    /**
+     * Helper method, since it's used very frequently in our code
+     *
+     * @param key
```

Comment:
shouldn't we specify the type of the parameters/return values? Can this be linted?









Or maybe we should use Typescript 🙈

Reply:
I guess this can be linted, but it's too early for the team